File: Interpreter C#/Interpreter/PrintStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class PrintStmt : IStmt
        {
            Exp exp;

            public PrintStmt(Exp e)
            {
                this.exp = e;
            }

            PrgState IStmt.execute(PrgState state)
            {
                MyIList<int> output = state.getOut();
                output.add(this.exp.eval(state.getSymTable()));
                return state;
            }

            public override string ToString()
            {
                return "print (" + this.exp.ToString() + ")";
            }
        }
    }
}

File: MAP/Interpretor2/src/Model/PrintStmt.java
package Model;
import Model.IStmt;
public class PrintStmt implements IStmt{
	public Exp exp;
	public PrintStmt(Exp exp)
	{
		this.exp=exp;
	}
	@Override
	public String toString(){
		return "print(" +exp.toString()+")";
	}
	public PrgState execute(PrgState state){
		MyIList<Integer> out=state.getOut();
		try{
		out.add(exp.eval(state.getSymTable(),state.getHeap()));}
		catch(Exception e)
		{
			System.out.println(e);
		}
		
		return state;
	}

}
